### PR TITLE
elementAt operator

### PR DESCRIPTION
**Pull request by rjhdby**
added reactiveX elementAt operator

### CODE REVIEW

File: lib/Rx/Observable.php [hunk 1]
```diff
@@ -1634,4 +1635,22 @@ public static function race(array $observables)
...
+     *
+     * @param int $index
+     * @return AnonymousObservable
+     *
+     * @demo elementAt/elementAt.php
+     * @demo elementAt/elementAtOverflow.php
+     * @operator
+     * @reactivex elementAt
+     */
+    public function elementAt($index=null){
```
Comment (davidwdan) [inline lib/Rx/Observable.php]: A couple of things here:
1. Why is `$index` optional?
2. There should be an optional default value parameter
3. nit pick:  there should be spaces around `=`

File: test/Rx/Functional/Operator/ElementAtTest.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            onCompleted(600),
+        ), $results->getMessages());
+    }
+
+    /**
+     * @test
+     */
+    public function take_zero_calls_when_index_greater_than_sequence_length()
+    {
+        $xs        = $this->createHotObservable(array(
```
Comment (davidwdan) [inline test/Rx/Functional/Operator/ElementAtTest.php]: nit: spacing

File: test/Rx/Functional/Operator/ElementAtTest.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Rx\Functional\Operator;
+
+
```
Comment (davidwdan) [inline test/Rx/Functional/Operator/ElementAtTest.php]: nit: extra line space

File: test/Rx/Functional/Operator/ElementAtTest.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+use Rx\Functional\FunctionalTestCase;
+use Rx\Observable\ReturnObservable;
+use Rx\Observer\CallbackObserver;
+
+
```
Comment (davidwdan) [inline test/Rx/Functional/Operator/ElementAtTest.php]: nit: extra space

File: test/Rx/Functional/Operator/ElementAtTest.php [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        $xs        = $this->createHotObservable(array(
+            onCompleted(220),
+        ));
+
+        $results = $this->scheduler->startWithCreate(function() use ($xs) {
+            return $xs->elementAt(3);
+        });
+
+        $this->assertMessages(array(
+            onCompleted(220),
```
Comment (davidwdan) [inline test/Rx/Functional/Operator/ElementAtTest.php]: Should emit an error if the index is out of range and no default is set

File: test/Rx/Functional/Operator/ElementAtTest.php [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        ), $results->getMessages());
+    }
+}
```
Comment (davidwdan) [inline test/Rx/Functional/Operator/ElementAtTest.php:90]: Can you add a test for when it disposes before the index is reached?

**Comment by rjhdby** [comment]
@davidwdan Ok. Get to work.

PS Does it exist some roadmap for RxPHP?

**Comment by davidwdan** [comment]
Most of the operators are added as needed.  As far as the implementation/tests go, we try to stay close to RxJS, with some deviations when it makes sense.

https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/elementat.md
https://github.com/Reactive-Extensions/RxJS/blob/master/tests/observable/elementat.js

**Comment by davidwdan** [comment]
@rjhdby we also try to keep the demos close to the RxJS examples, so we have a good baseline

**Comment by rjhdby** [comment]
@davidwdan Help me please, I don't understand one moment.
How I can see, test failed on
[x] demo/take/takeUntil.php

But in this PR I do not touch anything that can break this test. 

At last, in this branch takeUntil test work wrong. But I don't see why. Only difference from master is added function in Observable. All other changes is new files with ElementAt class and tests for it.

**Comment by davidwdan** [comment]
I reran the test and it passed, so it was probably just a timing issue.

**Comment by rjhdby** [comment]
@davidwdan I try find problem in takeUntil and find strange behavior. Sometimes it send 9 values, but sometime 10. Looks like bug when using Timer for generate sequence.

**Comment by davidwdan** [comment]
@mbonneau can you review this when you get a chance?

**Comment by mbonneau** [comment]
@rjhdby Can you reimplement this using `skip` and `take` right in the `Observable` class?

**Comment by mbonneau** [comment]
@rjhdby - Haven't heard back from you on this so I am closing this PR. If you would like to make a new PR, that would be welcome.

**Pull request closed without merging**